suite/tests/runall.cmake

```diff
@@ -212,6 +212,32 @@ elseif ("${nudge}" MATCHES "<attach>")
     kill_background_process(ON)
     message(FATAL_ERROR "*** ${nudge_cmd} failed (${nudge_result}): ${nudge_err}***\n")
   endif (nudge_result)
+elseif ("${nudge}" MATCHES "<detach>")
+  set(nudge_cmd run_in_bg)
+  string(REGEX REPLACE "<detach>"
+    "${toolbindir}/drrun@-attach@${pid}@-takeover_sleep@-takeovers@1000"
+    nudge "${nudge}")
+  string(REGEX REPLACE "@" ";" nudge "${nudge}")
+  execute_process(COMMAND "${toolbindir}/${nudge_cmd}" ${nudge}
+    RESULT_VARIABLE nudge_result
+    ERROR_VARIABLE nudge_err
+    OUTPUT_VARIABLE nudge_out
+    )
+  # combine out and err
+  set(nudge_err "${nudge_out}${nudge_err}")
+  if (nudge_result)
+    kill_background_process(ON)
+    message(FATAL_ERROR "*** ${nudge_cmd} failed (${nudge_result}): ${nudge_err}***\n")
+  endif (nudge_result)
+  do_sleep(5000)
```

Comment:
It would be better to wait on some output from the app so we know it's fully attached and we'll be doing a detach at the right point.  You could imagine a slow overloaded test VM still attaching after this 50ms.  Using sleeps like this tends to result in flaky tests.  In fact doesn't the script below here look for the "attach" and "thread init" output?  We want to wait for that before we detach, right?  This may be the source of your test failures.

Reply:
Thanks for the heads up, I have modified it following your suggestions. The workflow of the test  looks more reasonable.  But the vs2019-32 test always complain timeout. Is it possible to add it into ignore-failures list?

Reply:
Are you able to reproduce the 32-bit failure locally?

Reply:
After my last commit, vs2019-32 work well on my laptop.

Reply:
So it works locally, but hangs on GA (Github Actions).  One option is to use tmate to debug it on GA: https://github.com/marketplace/actions/debugging-with-tmate. It seems worth some effort to try to figure out why it's failing on GA, because it could indicate a real bug where detach for 32-bit doesn't work on a lot of setups and so users who try to use it might all hit hangs too.  If after some effort we can't figure it out then we can cut our losses and at least have the 64-bit test.  For the tmate what I would do is (temporarily) delete all the `.github/workflows/*.yml` files except the windows one and then edit that for tmate (and maybe to remove all the other tests).  (This assumes a PR from a clone will use the clone's yml files...I think so, once permissions are given by us to run our tests.)

Reply:
I have solved this problem by chance. I commit [some test codes](https://github.com/DynamoRIO/dynamorio/pull/6221/commits/5b4262dde3f9dcd2dea1b12634369de9e9b15a7a) to output the detach's result, it shows that the attach and detach operation can be completed successfully. I guess that the target process can't be exit smoothly, so I [kill the process manually](https://github.com/DynamoRIO/dynamorio/pull/6221/commits/127834b102eaeebca5f7f1ec57f0689a96198460) at the end of runall.cmake, and then all the tests can pass.